File: user_guides/webhooks/verify_webhook_samples/ruby/webhook.rb
#
# gem install sinatra awesome_print
# ruby webhook.rb
#

require "sinatra"
require "awesome_print" # pretty printing using `ap` instead of `puts`

post '/' do
  # return response success with no content
  status 204

  signature = request.env["HTTP_X_WEBHOOK_SIGNATURE"]
  payload = request.body.read
  security_token = "tandawebhooktest"
  actual_signature = OpenSSL::HMAC.hexdigest(OpenSSL::Digest.new("sha1"), security_token, payload)

  ap "Signature: #{signature}"
  ap "Actual Signature: #{actual_signature}"

  # Pretty print webhook payload
  ap payload
end

File: user_guides/webhooks/verify_webhook_samples/node/webhook.js
//
// npm init
// npm install express body-parser
// node webhook.js
//

const express = require('express');
const bodyParser = require('body-parser');
const crypto = require('crypto');
const util = require('util');

const ap = (data) => console.log(util.inspect(data, { colors: true, depth: null }));

const app = express();
const port = 3000;
const securityToken = 'tandawebhooktest';

app.use(bodyParser.text());

app.post('/', (req, res) => {
  const payload = req.body;
  const signature = req.headers['x-webhook-signature'];
  const actualSignature = crypto.createHmac('sha1', securityToken).update(payload).digest('hex');

  ap(`Signature: ${signature}`);
  ap(`Actual Signature: ${actualSignature}`);

  ap(payload);

  res.status(204).send();
});

app.listen(port, () => {
  console.log(`Server running on http://localhost:${port}`);
});
